frontend/src/metabase/redux/embedding-data-picker.ts

```diff
@@ -0,0 +1,59 @@
+import type { PayloadAction } from "@reduxjs/toolkit";
+import { createSlice } from "@reduxjs/toolkit";
+import _ from "underscore";
+
+import {
+  DEFAULT_EMBEDDING_ENTITY_TYPES,
+  type EmbeddingEntityType,
+} from "metabase/embedding-sdk/store";
+import type { EmbeddingDataPickerState } from "metabase-types/store/embedding-data-picker";
+
+export const DEFAULT_EMBEDDING_DATA_PICKER_STATE: EmbeddingDataPickerState = {
+  entityTypes: DEFAULT_EMBEDDING_ENTITY_TYPES,
+};
+
+const embeddingDataPickerSlice = createSlice({
+  name: "embeddingDataPicker",
+  initialState: DEFAULT_EMBEDDING_DATA_PICKER_STATE,
+  reducers: {
+    setEntityTypes: (
+      state,
+      action: PayloadAction<EmbeddingEntityType[] | undefined>,
+    ) => {
+      if (action.payload) {
+        const entityTypes = normalizeEntityTypes(action.payload);
+        if (!_.isEqual(state.entityTypes, entityTypes)) {
```

Comment:
ensure we don't rerender unnecessarily when the array value doesn't change even though the reference has changed.